### DISCUSSION TITLE
It just doesn't work

### DISCUSSION THREAD

**Discussion by shizenyasuya** [mrlt8/docker-wyze-bridge#663] [Q&A] [answered]
I'm showing my complete idiocy here, and this is likely more of a newb to docker issue than anything wrong with the project, but how do people get this to work in windows?  I've downloaded docker desktop (which supposedly includes docker compose) and I tried to run the following in command prompt and in windows power shell:

docker run \
  -e WYZE_EMAIL=you@email.com \
  -e WYZE_PASSWORD=yourpassw0rd \
  -p 1935:1935 -p 8554:8554 -p 8888:8888 -p 5000:5000 \
  mrlt8/wyze-bridge:latest

But I keep getting reference errors. I feel like this may be because I'm using windows instead of linux, but I would really like to run this on the same computer I use for blue iris, so it kind of needs to be windows.
I thought maybe it was the return characters trying to run each line as an individual command, so I tried removing it all as a single line and got basically the same response, 

"C:\Users\shize>docker run \
docker: invalid reference format.
See 'docker run --help'."

And each following line with email, password or port information just said that "-p" or "-e" is not recognized as an internal or external command.

where am I supposed to run these lines of script?  I feel like when I installed docker, the really poor tutorial loaded a very different shell that maybe would run the commands / script better? but the tutorial for docker in no way tells you how to get back to that shell ever again after the tutorial.

I've also tried creating the .yml file by copying the information into a word document and saving it as .yml instead of .txt, but then I have no idea where to put that file to do anything with it. Docker documentation says to put it in the project directory, but I have no idea where this project lives on my computer.

I did finally get the project downloaded using the pull command in command prompt and I even got the project to run, Docker desktop has a handy tool to declare enviornment variables so I was able to set my WYZE username and password that way, but the same tool has the audacity to tell me that the image doesn't expose any ports so I have no place to declare the ports using that tool, and in my mind, I'm like... the whole point of this image is to expose ports! what do you mean no ports are exposed!?!?!? 

Thank you to anybody who reads my whiny comment / plea for help / mercy.  In the end I should probably just get rid of my one wyze cam and replace it with something that supports RTSP out of the box, but idk I kind of wanted to try using docker because it seems to be very popular, but I'm so frustrated with trying to get it to work or do anything remotely useful.

**Comment by JA16122000** [comment]
I think, you have to install docker for windows first. 
https://docs.docker.com/desktop/install/windows-install/

Then you can go further with this WYZE project.

**Reply by shizenyasuya** [reply]
Lol. thank you. I did in fact get at least that part right. I had docker desktop installed before attempting this.

**Comment by cheme75** [comment]
I’d suggest uninstalling docker desktop, re-install 4.12 - get it from the docker GitHub release notes page. Latest version created problems on my pc. In win firewall, open the 4 ports listed. I created a folder such as c:\docker\bridge, I put my yml file in that folder then cd\ to that folder, then run docker-compose pull, then docker-compose up -d. If you use 2FA with wyze, then follow instructions for that too.

**Reply by shizenyasuya** [reply]
I had it working without downgrading to 4.12 for a minute, but I did eventually have to downgrade to 4.12 to get it to work reliably. This is a really good suggestion.

**Comment by mrlt8** [comment]
I think you're missing the `\` at the end of the lines:

```
docker run \
  -e WYZE_EMAIL=you@email.com \
  -e WYZE_PASSWORD=yourpassw0rd \
  -p 1935:1935 -p 8554:8554 -p 8888:8888 -p 5000:5000 \
  mrlt8/wyze-bridge:latest

```

Alternatively, you can try running everything in a single line:


```
docker run -e WYZE_EMAIL=you@email.com -e WYZE_PASSWORD=yourpassw0rd -p 1935:1935 -p 8554:8554 -p 8888:8888 -p 5000:5000 mrlt8/wyze-bridge:latest
```

**Reply by shizenyasuya** [reply]
You're right, my code above is missing the slashes, but that is a proofreading error on my part. They w ere there the several times I tried to run the command in commandprompt.

**Comment by shizenyasuya** [comment, marked as answer]
Thank you everybody who responded. I appreciate all your efforts and thoughts. I did eventually find, at least wtih docker desktop for windows. In the docker-compose.yml file, you have to format the environment variables slightly differently, no curly brackets, or dollar sign, just the string of your email and password. 

Original docker-compose.yml contents:
version: '2.4'
services:
    wyze-bridge:
        container_name: wyze-bridge
        restart: unless-stopped
        image: mrlt8/wyze-bridge:latest
        ports:
            - 1935:1935 # RTMP
            - 8554:8554 # RTSP
            - 8888:8888 # HLS
            - 5000:5000 # WEB-UI
        environment:
            - WYZE_EMAIL=${WYZE_EMAIL} # Replace with wyze email 
            - WYZE_PASSWORD=${WYZE_PASSWORD} # Replace with wyze password

What worked for me:

version: '2.4'
services:
    wyze-bridge:
        container_name: wyze-bridge
        restart: unless-stopped
        image: mrlt8/wyze-bridge:latest
        ports:
            - 1935:1935 # RTMP
            - 8554:8554 # RTSP
            - 8888:8888 # HLS
            - 5000:5000 # WEB-UI
        environment:
            - WYZE_EMAIL=email # Replace with wyze email 
            - WYZE_PASSWORD=password # Replace with wyze password

There was also a huge learning curb on my part with this being my first time using docker. I had thought I had to have the image file and the docker-compose.yml file in the same directory, but now, you just run docker-compose up from the directory containing the docker-compose.yml file, and it creates the container based on the image specified in the yml file. If someone had told me this last night, before I posed this thread, my mind would have been blown.  Anyway If anybody else is out there struggling with this, trying to get it to run on windows, I hope my experience can be beneficial to you. Docker is still a mystery (misery) to me, but I'm hoping to do more with home assistant eventually so fingers crossed, I can change that.

**Reply by vincom** [reply]
im a noob to docker. not a noob to pc's.
i got everything running perfect as to my 10 wyze cameras configured with blue iris.
i got docker-desktop to auto start on windows boot.
 i just want to get the docker-compose.yml file so i can edit it to make the wyze-bridge auto start on windows boot.
but where the f is docker-compose.yml.

actually where are the wyze bridge container files if i had to navigate to it. from terminal or explorer.
i can see all the files for the wyze bridge in docker-desktop but no .yml files

i read about docker compose but nobody really explains how it works with docker-desktop.

maybe i should have started a new discussion
